### ISSUE TITLE
Allow localisation scopes to be customised

### ISSUE THREAD

**Issue by peteryates** [DFE-Digital/govuk_design_system_formbuilder#80]
Now form builder configuration #73 and localisation #69 are merged into the version 1.1.0 branch, allowing for custom localisation scopes can be added should be fairly straightforward.

Currently the scope is hardcoded to:

```ruby
['helpers', context, @object_name, @attribute_name].join('.') 
```

Which expects a locales file in the following format:

```yaml

helpers:
  label:
    person:
      name: Who goes there?
      cv: Tell us about your employment history
      favourite_colour: What is your favourite colour?
      photo: Upload a recent passport photo
  fieldset:
    person:
      favourite_colour: To which colours are you most partial?
      born_on: What is your date of birth?
      projects: What do you do all day at work?
  hint:
    person:
      name: Enter your full name including middle names, nicknames and aliases
      favourite_colour: |-
        Yes, there are plenty of options but please pick the one you enjoy most
      cv: Where did you work? What did you do?
      photo: Make sure your face is clear and you are looking at the camera
      born_on: Check your birth certificate or passport
      projects: If you cannot remember check with your teammates or manager                                                                           
```

It would appear that the final two elements in the array `@object_name`, and `@attribute_name` would need to remain the same, but switching the beginning (in our case, `helpers` and `context`) should be possible.

There is an existing project that would benefit from this functionality, see [this PR](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/pull/90).

* [x] Add configuration options for customising the localisation schema, it should take an array of symbols
* [x] Ensure the customisation can work independently for _contexts_ (labels, hints and fieldsets)
* [x] Add tests that cover all the new functionality
* [x] Document the configuration

**Comment by peteryates** [comment]
cc @luke-slowen - will this do the trick for you guys?

**Comment by luke-slowen** [comment]
Sorry Peter, have you mentioned the right person? Or, if you are looking for an architecture perspective, can you be more specific in the ask please?

**Comment by peteryates** [comment]
Sorry, autocompletion failed :man_facepalming: 

@leeky

**Comment by leeky** [comment]
@peteryates Sounds great! This would allow us to configure the labels to use the ActiveRecord i18n scope and use your default for the others.

Thanks again :tada:

**Comment by peteryates** [comment]
@leeky Excellent

Just to keep you in the loop, there's now a [beta version of the gem](https://rubygems.org/gems/govuk_design_system_formbuilder/versions/1.1.0.beta.1) that comes with i18n, a huge refactor and some other goodies.

**Comment by peteryates** [comment]
Merged into the 1.1.0 branch

Documented [here](https://deploy-preview-76--govuk-form-builder.netlify.com/building-blocks/localisation/#customising-locale-structure), beta release [here](https://rubygems.org/gems/govuk_design_system_formbuilder/versions/1.1.0.beta.2) cc @leeky
